pilot/pkg/model/push_context.go

```diff
@@ -820,7 +829,11 @@ func (ps *PushContext) getExportedDestinationRuleFromNamespace(owningNamespace s
 // IsClusterLocal indicates whether the endpoints for the service should only be accessible to clients
 // within the cluster.
 func (ps *PushContext) IsClusterLocal(service *Service) bool {
-	_, ok := MostSpecificHostMatch(service.Hostname, ps.clusterLocalHosts)
+	m := make(map[host.Name]struct{}, len(ps.clusterLocalHosts))
```

Comment:
nit: should we allow passing unset map to MostSpecificHostMatch to avoid allocs here? clusterLocalHosts is always small and this is called many many times

Reply:
okay, I will fix it

Reply:
@howardjohn It’s being repaired in #28678